alf/algorithms/mcts_algorithm.py

```diff
@@ -580,15 +703,74 @@ def _backup(self, trees: _MCTSTree, search_paths, path_lengths, values):
             discounts = self._discount**(path_lengths.unsqueeze(0) - steps)
             discounted_return = values.unsqueeze(0) * discounts
 
-        value_sum = trees.value_sum[B, search_paths] + discounted_return
+        value_sum = trees.value_sum[B, search_paths] + count * discounted_return
 
         valid = depth < path_lengths
         nodes = (B.expand(T, -1)[valid], search_paths[valid])
-        trees.visit_count[nodes] += 1
+        trees.visit_count[nodes] += count
         trees.value_sum[nodes] = value_sum[valid]
         trees.update_value_stats((B, search_paths), valid)
         self._update_best_child(trees, nodes)
 
+    def _expand_children(self, trees: _MCTSTree, nodes, n, to_plays):
+        """Expand all children of nodes.
```

Comment:
Also state that the sum of the value due to the expansion will be returned (for computing the backing up)?

Reply:
added.